### PR TITLE
Zendesk Connector

### PR DESCRIPTION
**Pull request by sanjitr17**

### CODE REVIEW

File: connectors/zendesk.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+
+### 1. Upload attachment
+
+First, you need to upload the attachment. You can upload a text file by entering your text into the body field. Specify a filename including the file extension you want to use (txt, doc, csv). For text files, you can leave the Content type field blank.
+
+![Zendesk Attachment](/assets/images/connectors/zendesk/upload_attachment.png)
+*Upload Attachment*
+
+If you are transferring a file from another application, say Box, you will need to be able to provide the file's Body Content in this section. Here's an example of how you can do so : [1) JIRa -> Zendesk Attachment](https://www.workato.com/recipes/341176-new-issue-from-jira-will-search-for-tickets-in-zendesk-and-upload-attachment#recipe)
```
Comment (brandonz2892) [inline connectors/zendesk.md]: Capitalize the link "JIRa" into "JIRA".

File: connectors/zendesk.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+### 2. Update ticket
+
+In order to update the ticket, you must provide the attachment token from the upload attachment step output. Pass in this token to the Upload attachment token field. You will also need to provide the ID for the ticket you want to update along with a comment that the attachment will be uploaded with.
+
+![Zendesk Update Ticket](/assets/images/connectors/zendesk/update_ticket.png)
+*Update Ticket*
+
+## Zendesk update does not update the record if there is no change
+
+Zendesk is quite smart in how it handles updates. We have noticed that although we issued an update via the API, if the record was the same as before it does not generate a new trigger event. This is a good thing and helps in stopping update cascades going between two systems. The reason it is especially good is that zendesk does not expose the 'last updated by user' field and so there is no way to filter out updates made by the api.
```
Comment (brandonz2892) [inline connectors/zendesk.md]: Capitalize "api" in the end to "API".

File: connectors/zendesk.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+This is true of contact, haven't verified if all records behave the same way. Don't believe ticket (plus comments) behaves the same way.
```
Comment (brandonz2892) [inline connectors/zendesk.md]: Change to "This is true of contacts, and we haven't verified if all records behave the same way. We don't believe tickets do.

File: connectors/zendesk.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+## Zendesk Error Message: 422 Unprocessable Entity
```
Comment (brandonz2892) [inline connectors/zendesk.md]: Move this into a "Troubleshooting" section.

File: connectors/zendesk.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+### Solution: 
+Double check your data and make changes before restarting your recipe/re-running the job
+
+## Zendesk : Assigning user to Organization(s)
+
+### Allocating Organization membership
+Assigning New user/Reassigning existing user to Organisations
+
+One of the more prominent feature of Zendesk is the distribution of tasks based on workplace environment for each user. When performing syncs between Zendesk and other application, it is common that relocation of organisation is needed. This section will show you various techniques on how you may allocate users to different organizations for different workplace environment.
```
Comment (brandonz2892) [inline connectors/zendesk.md]: "prominent feature" should be plural.

File: connectors/zendesk.md [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+### 1) Assigning Organization during creation of the user
+
+You will be able to assign a newly created user to an organization based on the organisation ID or its name.This is done by providing the value in the organisation field in Create New User Step.
```
Comment (brandonz2892) [inline connectors/zendesk.md]: Change "based on the organisation ID or its name" to "based on the organization's ID or name". Change organisation to "organization".

File: connectors/zendesk.md [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+
+![Zendesk Assign Organization ID](/assets/images/connectors/zendesk/assign_organization_id.png)
+If you do not see the Organisation field, click on Add/Remove Optional field, and select Organisation ID
```
Comment (brandonz2892) [inline connectors/zendesk.md]: Change "organisation" to "organization".

File: connectors/zendesk.md [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+*Assign Organization ID*
+
+You may also toggle it by clicking on the small error beside the field to use organisation name instead. 
+
+![Zendesk Assign Organization Name](/assets/images/connectors/zendesk/assign_organization_name.png)
+*Assign Organization Name*
+
+### 2) Updating an exiting user's organisation
```
Comment (brandonz2892) [inline connectors/zendesk.md]: Change "organisation" to "organization".

File: connectors/zendesk.md [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+
+### 3) Create organization membership action
+If you have users that are enabled for multiple organisation membership, you may use the Create Organisation membership action to allocate the user to the organisation
+
+![Zendesk Create Organization Membership](/assets/images/connectors/zendesk/create_organization_membership.png)
+*Create Organization Membership*
+
+## Updating a batch of existing tickets
+
+There can be various situation where tickets of a certain type, belonging to one organisation, or applied of a certain tags, need to be updated. Whether is a reassignment, a broadcasting comment, or simply closing these tickets, updating each ticket one by one can be taxing upon your Zendesk API limits each day.
```
Comment (brandonz2892) [inline connectors/zendesk.md]: "Various situation" should be plural.

File: connectors/zendesk.md [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+
+With Bulk ticket updates action, you will be able to save on your API usage, as well as making your recipes look better more efficient. Here are a few simple steps on how you can do so.
```
Comment (brandonz2892) [inline connectors/zendesk.md]: Change "as well as making...more efficient" to "as well as make...more efficiently".

File: connectors/zendesk.md [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+
+### 1) Obtain a Tickets List Item Data Pill
+
+The Bulk Ticket Updates action requires a number of TicketID concatenated by commas in order to work. If you'd like to auto-generate a list of those tickets under a specific filter, simply perform a Search Tickets action.
```
Comment (brandonz2892) [inline connectors/zendesk.md]: Change "TicketID" to "Ticket IDs".

File: connectors/zendesk.md [hunk 12]
```diff
@@ -0,0 +1,155 @@
...
+
+![Zendesk Generate List](/assets/images/connectors/zendesk/generate_list.png)
+*Generate List*
+
+### 2) Listing tickets in a comma separated format
+
+With its output, you can now use Workato's formula mode to extract its Ticket ID in a comma separated structure. First. Click on the formula mode on beside the field.
```
Comment (brandonz2892) [inline connectors/zendesk.md]: Change "First. Click" to "First, click".

File: connectors/zendesk.md [hunk 13]
```diff
@@ -0,0 +1,155 @@
...
+*Generate List*
+
+### 2) Listing tickets in a comma separated format
+
+With its output, you can now use Workato's formula mode to extract its Ticket ID in a comma separated structure. First. Click on the formula mode on beside the field. 
+
+![Zendesk Formula Mode](/assets/images/connectors/zendesk/formula_mode.png)
+*Formula Mode*
+
+Then, simply drag the Tickets data pill from your Search ticket action. The Tickets datapill will be the one accompanied by a List symbol (one with multiple stripes).
```
Comment (brandonz2892) [inline connectors/zendesk.md]: Make "data pill" one word.

File: connectors/zendesk.md [hunk 14]
```diff
@@ -0,0 +1,155 @@
...
+
+With its output, you can now use Workato's formula mode to extract its Ticket ID in a comma separated structure. First. Click on the formula mode on beside the field. 
+
+![Zendesk Formula Mode](/assets/images/connectors/zendesk/formula_mode.png)
+*Formula Mode*
+
+Then, simply drag the Tickets data pill from your Search ticket action. The Tickets datapill will be the one accompanied by a List symbol (one with multiple stripes). 
+
+![Zendesk Tickets Data Pill](/assets/images/connectors/zendesk/tickets_data_pill.png)
+*Tickets Data Pill*
```
Comment (brandonz2892) [inline connectors/zendesk.md]: Make "data pill" one word.

File: connectors/zendesk.md [hunk 15]
```diff
@@ -0,0 +1,155 @@
...
+
+Next, we are going to use the .pluck() formula to assist us in getting the Ticket IDs of all of the tickets. Simply copy and paste the following Formula below and place it right after the data pill in the field : 
+
+![Zendesk TicketID Formula](/assets/images/connectors/zendesk/ruby_code.png)
+*TicketID Formula in Ruby*
+
+It should now look like this :  
+![Zendesk Inputted Formula](/assets/images/connectors/zendesk/inputted_formula.png)
```
Comment (brandonz2892) [inline connectors/zendesk.md]: Image doesn't load.

File: connectors/zendesk.md [hunk 16]
```diff
@@ -0,0 +1,155 @@
...
+
+Then, simply drag the Tickets data pill from your Search ticket action. The Tickets datapill will be the one accompanied by a List symbol (one with multiple stripes). 
+
+![Zendesk Tickets Data Pill](/assets/images/connectors/zendesk/tickets_data_pill.png)
+*Tickets Data Pill*
+
+Next, we are going to use the .pluck() formula to assist us in getting the Ticket IDs of all of the tickets. Simply copy and paste the following Formula below and place it right after the data pill in the field : 
+
+![Zendesk TicketID Formula](/assets/images/connectors/zendesk/ruby_code.png)
+*TicketID Formula in Ruby*
```
Comment (brandonz2892) [inline connectors/zendesk.md]: Separate "TicketID" to "Ticket ID".

File: connectors/zendesk.md [hunk 17]
```diff
@@ -0,0 +1,155 @@
...
+
+It should now look like this :  
+![Zendesk Inputted Formula](/assets/images/connectors/zendesk/inputted_formula.png)
+*Inputted Formula*
+
+### 3) Specify the fields to update
+With this, you can now specify the fields that you'd like to update with. In this example, we are closing all tickets that has tag 'completed'.
```
Comment (brandonz2892) [inline connectors/zendesk.md]: Change "tickets that has tag" to "tickets that have the tag".

File: connectors/zendesk.md [hunk 18]
```diff
@@ -0,0 +1,155 @@
...
+You have now successfully automated a long, repetitive work in only **two** simple Workato steps!
+
+![Zendesk Bulk Tickets Recipe](/assets/images/connectors/zendesk/bulk_tickets_final_recipe.png)
+*Bulk Tickets Final Recipe*
+
+## Zendesk : What is User Identity
+
+In Zendesk, each user may be assigned with multiple "identities", pieces of information or profiles that will put you in contact with the person. Examples of such user identities are : email, Twitter, Facebook, Google, etc.
+
+### List User Identities
```
Comment (brandonz2892) [inline connectors/zendesk.md]: Make "data pills" one word.

File: connectors/zendesk.md [hunk 19]
```diff
@@ -0,0 +1,155 @@
...
+In Workato, a user's identities may be retrieved by providing a user ID. The output of the action will contain a List Item named Identity, while a series of data pills will be made available, however the main output to focus on would be these :
```
Comment (brandonz2892) [inline connectors/zendesk.md]: Make "data pills" one word.

File: connectors/zendesk.md [hunk 20]
```diff
@@ -0,0 +1,155 @@
...
+
+In Zendesk, each user may be assigned with multiple "identities", pieces of information or profiles that will put you in contact with the person. Examples of such user identities are : email, Twitter, Facebook, Google, etc.
+
+### List User Identities
+In Workato, a user's identities may be retrieved by providing a user ID. The output of the action will contain a List Item named Identity, while a series of data pills will be made available, however the main output to focus on would be these :  
+
+![Zendesk List User Identities](/assets/images/connectors/zendesk/list_user_identities.png)
+*List User Identities*
+
+**Type** : the kind of identity that it is.
```
Comment (brandonz2892) [inline connectors/zendesk.md]: Capitalize "the".

File: connectors/zendesk.md [hunk 21]
```diff
@@ -0,0 +1,155 @@
...
+**Value** : What is the value contained in this Identity field.
```
Comment (brandonz2892) [inline connectors/zendesk.md]: Change "What is the value contained in this Identity field." to "The value contained in this Identity field."

File: connectors/zendesk.md [hunk 22]
```diff
@@ -0,0 +1,158 @@
+---
+title: Zendesk
+date: 2017-06-23 06:15:00 Z
+---
+
+# Zendesk
+
+## Zendesk search tickets action
```
Comment (nandithar) [inline connectors/zendesk.md]: Make a heading called "Actions" and then but "Zendesk search tickets action" under it

File: connectors/zendesk.md [hunk 23]
```diff
@@ -0,0 +1,158 @@
...
+
+![Zendesk List User Identities](/assets/images/connectors/zendesk/list_user_identities.png)
+*List User Identities*
+
+**Type** : The kind of identity that it is.
+**Value** : The value contained in this Identity field.
+
+If you'd like to populate all Identities to another application, remember to use a Repeat Action to go through every single User Identity available for this user. To learn more about how to use Repeat Action, check out this link for a detailed guide.
+
+## Troubleshooting
```
Comment (nandithar) [inline connectors/zendesk.md]: Delete troubleshooting section

File: connectors/zendesk.md [hunk 24]
```diff
@@ -0,0 +1,158 @@
...
+### List User Identities
+In Workato, a user's identities may be retrieved by providing a user ID. The output of the action will contain a List Item named Identity, while a series of datapills will be made available, however the main output to focus on would be these :  
+
+![Zendesk List User Identities](/assets/images/connectors/zendesk/list_user_identities.png)
+*List User Identities*
+
+**Type** : The kind of identity that it is.
+**Value** : The value contained in this Identity field.
+
+If you'd like to populate all Identities to another application, remember to use a Repeat Action to go through every single User Identity available for this user. To learn more about how to use Repeat Action, check out this link for a detailed guide.
```
Comment (nandithar) [inline connectors/zendesk.md]: no link

File: connectors/zendesk.md [hunk 25]
```diff
@@ -0,0 +1,158 @@
...
+![Zendesk Inputted Formula](/assets/images/connectors/zendesk/inputted_formula.png)
+*Inputted Formula*
+
+### 3) Specify the fields to update
+With this, you can now specify the fields that you'd like to update with. In this example, we are closing all tickets that have the tag 'completed'. 
+
+![Zendesk Status](/assets/images/connectors/zendesk/status.png)
+*Zendesk Status*
+
+You have now successfully automated a long, repetitive work in only **two** simple Workato steps!
```
Comment (nandithar) [inline connectors/zendesk.md]: "You have now successfully automated long, repetitive work in only **two** simple Workato steps!"

**Comment by eeshansim** [comment]
Outdated

**Pull request closed without merging**